packages/api/lib/writeRecords/write-granules.js

```diff
@@ -881,8 +845,21 @@ const writeGranulesFromMessage = async ({
   // so that they can succeed/fail independently
   const results = await Promise.allSettled(granules.map(
     async (granule) => {
-      // compute granule specific data.
-      const files = await _generateFilesFromGranule({ granule, provider });
+      // FUTURE: null files are currently not supported in update payloads
+      // RDS Phase 3 should revise logic to accept an explicit null value
+      if (granule.files === null) {
+        throw new CumulusMessageError('granule.files must not be null');
+      }
+
+      // This is necessary to set properties like
+      // `key`, which is required for the PostgreSQL schema. And
+      // `size` which is used to calculate the granule product
+      // volume
+      const files = granule.files ? await FileUtils.buildDatabaseFiles({
+        s3: s3(),
+        providerURL: buildURL(provider),
+        files: granule.files,
+      }) : undefined;
```

Comment:
We still need to check granule.files here. `buildDatabaseFiles` will return an empty array if granule.files is undefined and that will trigger removal of existing files.

Reply:
Note that this relies on the `null` check above. I could make this more explicit `granule.files === undefined`